server/src/db/repositories/reportRepository.js

```diff
@@ -30,6 +31,10 @@ const repository = () => {
     ReportModel.aggregate(sortedAggregation).exec(callback);
   };
 
+  const aggregateBuildingBlocks = (callback) => {
+    ReportModel.aggregate(getBuildingBlockListPipeline()).exec(callback);
```

Comment:
Just a note but we should not rely on pipelines in every case. For compatibility aggregations it's unavoidable but Gettling list of unique bbs and relevant software seems more simplistic and perhaps we should put more focus on utilization of [mongoose queries](https://mongoosejs.com/docs/api/query.html) or [aggregations ](https://mongoosejs.com/docs/api/aggregate.html). Which are much more readable than json schema pipelines.

Reply:
changed it to 
```
    const aggregation = ReportModel.aggregate([]);
    aggregation.group({
      _id: '$buildingBlock',
      softwares: { $addToSet: '$testApp' },
    });
    aggregation.project({
      _id: 1,
      supportedSoftwares: { $size: '$softwares' },
    });
    aggregation.sort({ _id: 1 }).exec(callback);
```
is that what you meant?